pubspec.yaml

```diff
@@ -9,10 +9,10 @@ environment:
   sdk: '>=2.4.0 <3.0.0'
 
 dependencies:
-  charcode: ^1.1.0
-  collection: '>=1.1.0 <2.0.0'
-  string_scanner: '>=0.1.4 <2.0.0'
-  source_span: '>=1.0.0 <2.0.0'
+  charcode: ^1.1.3
+  collection: ^1.14.12
+  string_scanner: ^1.0.5
+  source_span: ^1.6.0
```

Comment:
Can't do this without breaking support for old SDKs!

Reply:
Specifying the third semver number is an antipattern.

We can probably simplify these constraints since I suspect only the latest major version of each support Dart 2.

```yaml
  charcode: ^1.1.0
  collection: ^1.14.0
  string_scanner: ^1.0.0
  source_span: ^1.4.0
```

I don't see value in trying to limit further than that.